### PR TITLE
add keyball directory

### PR DESCRIPTION
**Pull request by Yowkees**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Yowkees** [comment]
I want to add my new keyborad with trackball "keyball". Thank you.

File: keyboards/keyball/keyball.c [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "keyball.h"
```
Comment (drashna) [inline keyboards/keyball/keyball.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keyball/keyball.h [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#pragma once
```
Comment (drashna) [inline keyboards/keyball/keyball.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keyball/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+      currentReport = pointing_device_get_report();
+      if (record->event.pressed) {
+        currentReport.buttons |= MOUSE_BTN3;
+      }
+      else {
+        currentReport.buttons &= ~MOUSE_BTN3;
+      }
+      pointing_device_set_report(currentReport);
+      return false;
+  }
```
Comment (drashna) [inline keyboards/keyball/keymaps/default/keymap.c:153]:
```suggestion
#ifndef MOUSEKEY_ENABLE
    if (IS_MOUSEKEY_BUTTON(keycode)) {
        report_mouse_t currentReport = pointing_device_get_report();
        if (record->event.pressed) {
            currentReport.buttons |= 1 << (keycode - KC_MS_BTN1);
        } else {
            currentReport.buttons &= ~(1 << (keycode - KC_MS_BTN1));
        }
        pointing_device_set_report(currentReport);
        pointing_device_send();
    }
#endif
```

File: keyboards/keyball/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  return true;
+}
+	
+/*#ifndef MOUSEKEY_ENABLE
```
Comment (drashna) [inline keyboards/keyball/keymaps/default/keymap.c:157]: Ah, I see you had it.  This is probably easier, though.

File: keyboards/keyball/rev1/rev1.h [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#pragma once
```
Comment (drashna) [inline keyboards/keyball/rev1/rev1.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/keyball/rev1/rev1.c [hunk 6]
```diff
@@ -0,0 +1 @@
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/keyball/rev1/rev1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]
If you haven't already, you may want to check out my dactyl manuform code, as it includes a split transport and additional code that supports using the pointing device no matter which side is plugged in. 

https://github.com/qmk/qmk_firmware/tree/master/keyboards/handwired/dactyl_manuform/5x6_right_trackball

**Pull request closed without merging**